### ISSUE TITLE
Create Inventory Item from Asset fails if sync_hardware_serial_asset_tag = True

### ISSUE THREAD

**Issue by Starkstromkonsument** [ArnesSI/netbox-inventory#80]
If you set `sync_hardware_serial_asset_tag = True` and try to create an Inventory Item from an Asset using the button 
![grafik](https://user-images.githubusercontent.com/53019449/230728502-2da4782c-df81-4fd7-84ff-3924abfd089f.png)
on the Assets page, the action fails showing this error message:
![grafik](https://user-images.githubusercontent.com/53019449/230728142-44042416-1cb0-40b5-b038-959520679584.png)

IMHO this is not the expected behavior.

**Comment by matejv** [comment]
With the latest code and if I create new assets, I cannot reproduce this. 

I suspect the problem is with empty asset_tag fields values on assets. There was a bug there where empty values were stored as empty strings (`""`), which netbox didn't like so they have to be stored as `None`. I'm not sure but it's possible you created assets on the buggy version of the plugin.

Could you check what is the asset_tag field set on one of the assets that has a problem? In [netbox shell](https://demo.netbox.dev/static/docs/administration/netbox-shell/) run:

```py
from netbox_inventory.models import Asset
asset_id = id of asset here
asset = Asset.objects.get(id=asset_id)
print(repr(a.asset_tag))
```

This should print None (which is correct value) or `''` (which is problematic).

Alternatively try to edit the asset which causes trouble via the web interface: just click Edit and Save without changing anything. This should save an empty asset_tag values as None. Then try to create inventory item again and it should work.

If we confirm this is indeed the problem, we can then figure out how to fix your existing data.

**Comment by Starkstromkonsument** [comment]
Hi, thank you for your feedback. netbox shell returns this for the asset in question:

```
>>> print(repr(asset.serial))
None
>>> print(repr(asset.asset_tag))
None
```
So `None` is printed correctly.

I tried this:

1. create a new Asset (Serial Number and Asset Tag not empty)
2. create an Inventory Item from the new Asset
3. try to edit the Asset or the related Inventory Item

Everything works fine. So yes, the bug seems to be fixed in the current version.

> Alternatively try to edit the asset which causes trouble via the web interface: just click Edit and Save without changing anything. This should save an empty asset_tag values as None. Then try to create inventory item again and it should work.

This did not work for me. For my broken Asset I tried this instead:

1. modify Serial and Asset Tag on the Asset (both from an empty string to a test string)
2. Save the Asset
3. try to edit the related Inventory Item

Everything works fine for this Asset now. The error must be at a different place. As I solved the problem manually now, I can not reproduce it anymore.

But now a new Problem occurs: If I try to set the Serial Number of the Asset to an empty string it fails with the error message

>  Cannot change inventoryitem serial and asset tag if asset is assigned. Please update via inventory > asset instead.

now. This is reproducible on a freshly created Asset.

I tried something different:

1. Create a new Asset with empty Serial Number and Empty Asset Tag
2. Try to create an Inventory Item from the new Asset
3. It Fails with 

> Cannot change inventoryitem serial and asset tag if asset is assigned. Please update via inventory > asset instead.

Debugging output:

```
>>> asset_id = 25
>>> asset = Asset.objects.get(id=asset_id)
>>> print(repr(asset.serial))
None
>>> print(repr(asset.asset_tag))
None
```

Try the same thing with a Serial Number set on the new Asset and the error does not occur.

**Comment by matejv** [comment]
Thanks for the detailed troubleshooting. I hadn't noticed the code does not handle assets with a blank serial number correctly. I've fixed that in #85 

It should hopefully fix all your problems without any need to fix your existing data.

I'll publish a new release shortly.
